### ISSUE TITLE
How is batch norm computed on multiple GPUs?

### ISSUE THREAD

**Issue by martinpopel** [tensorflow/tensor2tensor#1070]
Does batch norm in T2T compute the mean and variance over all GPUs or for each GPU independently?

I am analyzing why scaling up `batch_size` from 1500 to 2000 [had no effect on BLEU](https://ufal.mff.cuni.cz/pbml/110/art-popel-bojar.pdf#page=16) in my experiments, but scaling up the number of GPUs from 1 to 2 (and thus the effective batch size from 1500 to 3000 ) [helped greatly](https://ufal.mff.cuni.cz/pbml/110/art-popel-bojar.pdf#page=20) when considering  BLEU after a given amount of training data.

**Comment by martinpopel** [comment]
Oops, I confused batch norm with layer norm. Transformer uses layer norm by default (and in my experiments), which is computed for each training example, so there is no issue/question with multi-GPU training and I am closing this issue.
Nevertheless, if someone can explain the different behavior in my experiments (when scaling batch_size vs. when scaling the number of GPUs), please let me know.

**Comment by gzhcv** [comment]
Hello. t2t set the hparams ' --sync=False ' by default.  Your expression shows that your experiment set the hparams ' --sync=True' ?  I guess the  experiment in Google's paper set the --sync=False. If the performance imporved with the batch_size growth, why Google don't set the --syc=True?
